doc/source/guide/in_depth/functional_guide.rst

```diff
@@ -0,0 +1,189 @@
+
+From a mathematical perspective, the above is a valid definition of a functional.
+However, since these functionals are primarily to be used for solving optimization problems, the following assumptions are made:
+
+ * the vector space :math:`X` is a Hilbert space.
+ * the field of scalars :math:`F` are the real numbers.
+
+The first assumption is made in order to simplify the concept of *convex conjugate functional*, see ``convex_conj`` under :ref:`implementation` for more details, or the Wikipedia articles on `convex conjugate`_ and `Legendre transformation`_.
+
+The second assumption is made in order to guarantee that we use a well-ordered set (in contrast to e.g. the complex numbers) over which optimization problems can be meaningfully defined, and that optimal solutions are in fact obtained.
+See, for example, the Wikipedia articles on `field`_, `ordered field`_ and `least-upper-bound property`_.
+
+Note that these conditions are not explicitly checked.
+However, using the class in violation to the above assumptions might lead to unknown behavior since some of the mathematical results might not hold.
+Also note that most of the theory, and most solvers, requires the functional to be convex.
+However this property is not stored or checked anywhere in the class.
+It is therefore the users responsibility to ensure that a functional has the properties required for a given optimization method.
+
+The intended use of the `Functional` class is, as mentioned above, to be used when formulating and solving optimization problems.
+One main difference with the `Operator` class is thus that it contains notions specially intended for optimization, such as *convex conjugate functional* and *proximal operator*.
+For more information on these concepts, see ``convex_conj`` and ``proximal`` under :ref:`implementation`.
+There is also a certain type of arithmetics associated with functionals, for more on this see :ref:`arithmetic`.
+
+
+.. _convex conjugate: https://en.wikipedia.org/wiki/Convex_conjugate
+.. _Legendre transformation: https://en.wikipedia.org/wiki/Legendre_transformation
+
+.. _field: https://en.wikipedia.org/wiki/Field_(mathematics)
+.. _ordered field: https://en.wikipedia.org/wiki/Ordered_field
+.. _least-upper-bound property: https://en.wikipedia.org/wiki/Least-upper-bound_property
+
+.. _implementation:
+
+Implementation of functionals
+=============================
+
+To define your own functional, start by writing::
+
+    class MyFunctional(odl.solvers.Functional):
+
+        """Docstring goes here."""
+    
+        def __init__(self, space):
+            # Sets `Operator.domain` to `space` and `Operator.range` to `space.field`
+            odl.solvers.Functional.__init__(self, space)
+
+        ...
+
+`Functional` needs to be provided with a domain, from which it infers the range.
+
+.. automethod:: Functional.domain
+
+Moreover, there are two optional parameters associated with a functional.
```

Comment:
" associated with a functional." is a bit open ended. Perhaps be more explicit and state "that can be provided in the initializer"